miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -305,4 +331,8 @@ export.finalize_transaction
     # assert no net creation or destruction of assets over the transaction
     exec.memory::get_input_vault_root exec.memory::get_output_vault_root assert_eqw.err=ERR_EPILOGUE_ASSETS_DONT_ADD_UP
     # => [OUTPUT_NOTES_COMMITMENT, FINAL_ACCOUNT_HASH]
+
+    exec.get_absolute_expiration_block movdn.8
+
+    exec.::std::sys::truncate_stack
```

Comment:
I wasn't sure why I was getting an address overflow error after executing (since some lines above, half of the stack is dropped anyway), but adding this fixed it so I was unblocked.

Reply:
I also wasn't sure what ordering we would prefer on the stack so I pushed it near the end until I had confirmation of what would be better.

Reply:
I think the stack order you have upon return is fine (i.e., `[OUTPUT_NOTES_COMMITMENT, FINAL_ACCOUNT_HASH, tx_expiration_block_num]`) - but let's update docs on line 253 above to reflect this.

As for truncating the stack, the reason why it happens is that `get_absolute_expiration_block` pushes an item onto the stack which already has 16 items. So, at the end, we end with 17 items, which results in an overflow. (the `swapdw dropw dropw` reduces stack depth to 16, because stack depth cannot drop below 16).

However, I would probably not use `std::sys::truncate_stack` here. Instead, the following would work better (i.e., would take much fewer cycles):
```
swapdw
exec.get_absolute_expiration_block
swap drop swapdw
```

Separately, I'm actually wondering if we need `swapdw dropw dropw` on line 331. Could you try removing these and see if things still work? If it doesn't work, we enter this procedure with a dirty stack and we should probably understand why. In this case, let's create an issue for this.

Reply:
Seems like it does not work without the `swapdw dropw dropw`.

Reply:
Let's create an issue to investigate this in the future.

Reply:
Created https://github.com/0xPolygonMiden/miden-base/issues/902